File: src/full_searching/carpet.py
# https://programmers.co.kr/learn/courses/30/lessons/42842
def solution(brown, red):
    red_height = 1
    red_width = red
    while red_height <= red_width:
        if (red_height + red_width) * 2 + 4 == brown:
            return [red_width + 2, red_height + 2]
        red_height += 1
        while red_height <= red and red % red_height != 0:
            red_height += 1
        red_width = red // red_height
    return [red_width + 2, red_height + 2]


print(solution(10, 2))
print(solution(8, 1))
print(solution(24, 24))

File: src/full_searching/Carpet.java
package src.full_searching;

import java.util.Arrays;

// https://programmers.co.kr/learn/courses/30/lessons/42842
public class Carpet {
    private static void getWh(int []fullWh, int []redWh, int red){
        redWh[0] = red / redWh[1];
        fullWh[0] = redWh[0] + 2;
        fullWh[1] = redWh[1] + 2;
    }
    public static int[] solution(int brown, int red) {
        int []fullWh = {0,0};
        int []redWh = {red,1};
        int area = brown + red;
        while(redWh[1] != red && fullWh[0] >= fullWh[1]){
            if(red % redWh[1] == 0){
                getWh(fullWh,redWh,red);
                if(fullWh[0] * fullWh[1] == area)
                    break;
            }
            redWh[1]++;
        }
        if(fullWh[0]==0 && fullWh[1] == 0){
            fullWh[0] = 3; fullWh[1] = 3;
        }
        return fullWh;
    }
    public static void main(String[] args) {
        int brown = 50;
        int red = 50;
        System.out.println(Arrays.toString(solution(brown, red)));
    }
}
